lib/services/shared_preferences/service.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+/// Service for interacting with [SharedPreferences].
+class SharedPreferencesService {
+  /// String key associated with the persisted accepted T&C version.
+  static const _tacAcceptedVersionKey = 'tac:accepted_version';
+
+  /// Wrapped instance.
+  final SharedPreferences _prefs;
+
+  const SharedPreferencesService(this._prefs);
+
+  /// Currently accepted shared preferences.
+  String? get termsAndConditionsAcceptedVersion => _prefs.getString(_tacAcceptedVersionKey);
+
+  /// Sets or deletes the currently accepted shared preferences.
+  void updateTermsAndConditionsAcceptedVersion(String? v) {
+    if (v == null) {
+      _prefs.remove(_tacAcceptedVersionKey);
+    } else {
+      _prefs.setString(_tacAcceptedVersionKey, v);
+    }
```

Comment:
```suggestion
  Future<void> updateTermsAndConditionsAcceptedVersion(String? v) async {
    if (v == null) {
      await _prefs.remove(_tacAcceptedVersionKey);
    } else {
      await _prefs.setString(_tacAcceptedVersionKey, v);
    }
```

IMHO we should await this and all cases where it is used should be made as futures
- userAccepted in TermsAndConditionAcceptance
- resetAccepted in TermsAndConditionAcceptance
- _onAcceptButtonPressed in _TermsAndConditionsScreenState